### PR TITLE
Support `@examplesIf` tag in `roxygen2`

### PR DESCRIPTION
**Pull request by vincentarelbundock**
The `roxygen2` package supports an `@examplesIf` tag. Next to that tag, users include R code that gets evaluated to `TRUE` or `FALSE`, which determines if the examples get evaluated.

When the documentation is rendered, `roxygen2` adds a weird long string (see below). [This is done here.](https://github.com/r-lib/roxygen2/blob/db4dd9a4de2ce6817c17441d481cf5d03ef220e2/R/rd-examples.R#L20)

This PR uses ugly/hacky regexes to extract the code to be evaluated form the `.Rd` file, run it, and decide if we should eval the code examples. The main problem with this approach is that if `roxygen2` changes the string, we'll need to update `altdoc` so that the regex works again.

What do you all think?

For reference, the `roxygen2` code looks like this:

```r
#' @export
#' @examplesIf on_workbench()
#'
... example code ...
#'
```

.Rd file looks like this:

```r
\examples{
\dontshow{if (on_workbench()) (if (getRversion() >= "3.4") withAutoprint else force)(\{ # examplesIf}

... example code ...

\dontshow{\}) # examplesIf}
}
```

### CODE REVIEW

**Review by etiennebacher — COMMENTED** [review]
Can you add some test for this? I think the easiest way to do that would be to add a `.R` file and its `.Rd` in `tests/testthat/examples/testpkg.altdoc/man` with some easy `@examplesIf 1+1 == 2` or something like that

File: tests/testthat/examples/testpkg.altdoc/R/examplesIf_false.R [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#' Examples If FALSE
+#'
+#' @param x A parameter
+#'
+#' @return Some value
+#' @export
+#'
+#' @examplesIf 2 + 2 == 5
+#' @examples
```
Comment (etiennebacher) [inline tests/testthat/examples/testpkg.altdoc/R/examplesIf_false.R:9]: You can remove the `@examples` tag

File: tests/testthat/examples/testpkg.altdoc/R/examplesIf_true.R [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#' Examples If TRUE
+#'
+#' @param x A parameter
+#'
+#' @return Some value
+#' @export
+#'
+#' @examplesIf 2 + 2 == 4
+#' @examples
```
Comment (etiennebacher) [inline tests/testthat/examples/testpkg.altdoc/R/examplesIf_true.R:9]: same

**Review by etiennebacher — COMMENTED** [review]
Can you also add two `expect_snapshot()` for those files in `test-render_docs.R`?

File: tests/testthat/examples/testpkg.altdoc/R/examplesIf_true.R (hunk 2 above)
Comment (vincentarelbundock) [reply tests/testthat/examples/testpkg.altdoc/R/examplesIf_true.R]: https://github.com/etiennebacher/altdoc/pull/271/commits/86fc1edef7f09f3106c77b0cfac6eced41b48818

File: tests/testthat/examples/testpkg.altdoc/R/examplesIf_false.R (hunk 1 above)
Comment (vincentarelbundock) [reply tests/testthat/examples/testpkg.altdoc/R/examplesIf_false.R]: https://github.com/etiennebacher/altdoc/pull/271/commits/86fc1edef7f09f3106c77b0cfac6eced41b48818

**Comment by vincentarelbundock** [comment]
> Can you also add two `expect_snapshot()` for those files in `test-render_docs.R`?

I don`t know how to. I tried adding the two lines in that file, but when I run `testthat::test_file()` the new snapshots are not created.

**Comment by etiennebacher** [comment]
Feel free to merge if you're happy with that, I don't have more comments

**Comment by vincentarelbundock** [comment]
Cool. FYI, I don't plan to make more changes to this, unless you make more requests. So don't wait on me to merge if you think it's fine.

**Comment by vincentarelbundock** [comment]
Haha, ok, we posted at the same time

**Pull request merged**